draco/analysis/flagging.py

```diff
@@ -997,6 +997,253 @@ def process(
+            [
+                ii
+                for ii, ax in enumerate(stream.vis.attrs["axis"])
+                if ax not in ["freq", "time", "ra"]
+            ]
+        )
+
+        chisq = stream.vis[:].real
+        weight = stream.weight[:]
+
+        wsum = np.sum(weight, axis=axsum)
+        chisq = np.sum(weight * chisq, axis=axsum) * tools.invert_no_zero(wsum)
+
+        # Gather all frequencies on all nodes
+        chisq = chisq.allgather()
+        wsum = wsum.allgather()
+
+        mask_input = wsum == 0.0
+
+        # Determine what time samples should be ignored when constructing the mask
+        if multiple_days:
+            mask_daytime = np.zeros(timestamp.size, dtype=bool)
+        else:
+            mask_daytime = self._day_flag_hook(timestamp)
+
+        mask_time = self._source_flag_hook(timestamp, freq)
+
+        mask = mask_input | mask_time
+
+        # Mask using median over time
+        mask_output = np.zeros((freq.size, timestamp.size), dtype=bool)
+
+        if self.nsigma_1d > 0.0:
+            mask_1d = self.mask_1d(chisq, mask | mask_daytime)[:, np.newaxis]
+            mask |= mask_1d
+            mask_output |= mask_1d
+
+        # Mask using dynamic spectrum
+        if self.nsigma_2d > 0.0:
+            # The inverse variance of the chisq per dof test statistic is ndof / 2.
+            # This expression assumes ndof is stored in the weight dataset.
+            w = ~mask * wsum / 2.0
+            mask_2d = self.mask_2d(chisq, w)
+
+            mask_output |= mask_2d & ~mask_daytime
+
+        # Save to output container
+        OutputContainer = (
+            containers.SiderealRFIMask
+            if "ra" in stream.index_map
+            else containers.RFIMask
+        )
+        output = OutputContainer(axes_from=stream, attrs_from=stream)
+
+        output.mask[:] = mask_output
+
+        return output
+
+    def mask_1d(self, y, m):
+        """Mask frequency channels where median chi-squared deviates from neighbors.
+
+        Parameters
+        ----------
+        y : np.ndarray[nfreq, ntime]
+            Chi-squared per degree of freedom.
+        m : np.ndarray[nfreq, ntime]
+            Boolean mask that indicates which samples to ignore
+            when calculating the median over time.
+
+        Returns
+        -------
+        mask : np.ndarray[nfreq]
+            Boolean mask that indicates frequency channels where
+            the median chi-squared over time deviates significantly
+            from that of the neighboring channels.
+        """
+        # For each frequency, caculate a weighted median
+        y = np.ascontiguousarray(y.astype(np.float64))
+        w = np.ascontiguousarray((~m).astype(np.float64))
+
+        med_y = weighted_median.weighted_median(y, w)
+        med_m = np.all(m, axis=-1)
+        med_w = np.ascontiguousarray((~med_m).astype(np.float64))
+
+        # Estimate a baseline
+        baseline = tools.arPLS_1d(med_y, mask=med_m, scale=self.reg_arpls)
```

Comment:
```suggestion
        baseline = tools.arPLS_1d(med_y, mask=med_m, lam=self.reg_arpls)
```

This is on me, I changed this variable name in `arPLS_1d` to better reflect the standard variables used in the equation it's solving